src/bp/core/services/dialog/dialog-engine.ts

```diff
@@ -223,6 +230,67 @@ export class DialogEngine {
     }
   }
 
+  private copyVarsToParent(flow: FlowView, currentNode: FlowNode, event: IO.IncomingEvent) {
+    const { workflows } = event.state.session
+    const { createVariable } = event.state
+
+    const childFlowVars = workflows[currentNode.flow!.replace('.flow.json', '')]?.variables
+    const outputVars = flow.variables?.filter(x => x.isOutput)
+
+    if (!outputVars || !childFlowVars) {
+      return
+    }
+
+    outputVars.forEach(v => {
+      const ouputVariableName = currentNode.subflow?.out?.[v.name]
+      if (!ouputVariableName) {
+        return
+      }
+
+      const cv = childFlowVars[v.name] as BoxedVariable<any>
+      if (cv) {
+        createVariable(ouputVariableName, cv.value, cv.type)
+      }
+    })
+  }
+
+  private sendVarsToChild(
+    flow: FlowView,
+    nextFlowName: string,
+    event: IO.IncomingEvent,
+    inputs?: { [variable: string]: SubWorkflowInput }
+  ) {
+    const { workflow, createVariable } = event.state
+
+    const inputVars = flow.variables?.filter(x => x.isInput)
+
+    if (!inputVars || !workflow.variables) {
+      return
+    }
+
+    inputVars.forEach(v => {
+      const input = inputs?.[v.name]
+      if (!input) {
+        return
+      }
+
+      let value: string
+      if (input.source === 'variable') {
+        const variable = workflow.variables[input.value]
+        if (!variable) {
+          return
+        }
+        value = variable.value
+      } else if (input.source === 'hardcoded') {
+        value = input.value
+      } else {
+        return
+      }
+
+      createVariable(v.name, value, v.type, { nbOfTurns: 10, specificWorkflow: nextFlowName })
```

Comment:
This could be simplified a bit:

```js
let value = input.value
if(input.source === 'variable'){
  value = workflow.variables[input.value]?.value
}

if(value !== undefined){
  createVariable
}
```

Reply:
The code was a bit bizarre because I wanted to allow setting the variable to the value undefined. I agree this is cleaner but it also prevents undefined from being passed as a parameter

Reply:
We shouldn't pass undefined around. If the goal is to delete a variable, let's think about it. In any case, if we want to do something to clear a value, I'd prefer we use "null" instead of undefined